src/asset/MountArchive.cpp

```diff
@@ -0,0 +1,89 @@
+#include "MountArchive.h"
+
+#include "bit7z/bitarchivereader.hpp"
+#include "utils/Log.h"
+
+#include <cstddef>
+#include <filesystem>
+#include <format>
+#include <string>
+#include <vector>
+
+MountArchive::MountArchive(std::filesystem::path archive_path)
+    : Mount(archive_path), reader(new bit7z::BitArchiveReader(library, path.string())) {
+}
+
+MountArchive::~MountArchive() {
+    MountArchive::save_cache();
+}
+
+Mount::MountType MountArchive::get_type() {
+    return ARCHIVE;
+}
+
+bool MountArchive::load() {
+    reset_reader();
+
+    try {
+        reader = new bit7z::BitArchiveReader(library, get_path().string());
+        reader->test();
+    }
+    catch (const std::exception& exception) {
+        return false;
+    }
+
+    load_cache();
+
+    return true;
+}
+
+bool MountArchive::contains_file(std::string path) {
+    if (static_cache.contains(path)) {
+        return true;
+    }
+    return false;
+}
+
+std::vector<std::byte> MountArchive::fetch_data(std::string path) {
+    uint32_t index = static_cache.at(path);
+    std::vector<std::byte> buffer;
+
+    try {
+        reader->extractTo(buffer, index);
+    }
+    catch (const std::exception& exception) {
+        Log::log_print(ERR,
+                       std::format("Failed to load data due to the following errror {}", exception.what()).c_str());
+    }
+
+    return buffer;

```

Comment:
what are the contents of buffer if we caught an exception? does the caller have a way of knowing that this data is bad?